pulsar-broker/src/test/java/org/apache/pulsar/client/impl/ProducerSemaphoreTest.java

```diff
@@ -66,7 +66,7 @@ public void testProducerSemaphoreInvalidMessage() throws Exception {
         ProducerImpl<byte[]> producer = (ProducerImpl<byte[]>) pulsarClient.newProducer()
                 .topic("testProducerSemaphoreAcquire")
                 .maxPendingMessages(pendingQueueSize)
-                .enableBatching(false)
+                .enableBatching(true)
```

Comment:
Should we also cover the case when batching is disabled? Maybe add a parameter in the test method.

Reply:
@Jason918 Add test to cover the batching is disabled case, PTAL Thx.